picamera2/picamera2.py

```diff
@@ -148,8 +211,15 @@ def __init__(self, camera_num=0, verbose_console=None, tuning=None):
                 os.environ["LIBCAMERA_RPI_TUNING_FILE"] = tuning_file.name
         else:
             os.environ.pop("LIBCAMERA_RPI_TUNING_FILE", None)  # Use default tuning
-        self.camera_manager = libcamera.CameraManager.singleton()
+        if not Picamera2._cm:
+            Picamera2._cm = CameraManager()
+        self.notifyme_r, notifymew = os.pipe()
```

Comment:
I wonder if we should be closing these file descriptors in our close method? I know a process can have lots of open file descriptors so it probably doesn't matter too much, but my sense of tidiness is feeling slightly uncomfortable...

Reply:
Also doing the following too
```
            self.notifymeread.close()
            os.close(self.notifyme_w)
```

Reply:
It seems trickier than I thought to call self.notifymeread.close() without issues.

Currently getting -
QSocketNotifier: Invalid socket 3 and type 'Read', disabling...

Which I'm just digging into.

Reply:
Managed to solve this as far as I understand by adding - parent, preview, tup = (None, None, None) to the MonitorThread